kraken/node_actions/common_node_functions.py

```diff
@@ -30,36 +33,83 @@ def get_node(node_name, label_selector, instance_kill_count):
     return nodes_to_return
 
 
-# Wait till node status becomes Ready
+# Wait until the node status becomes Ready
 def wait_for_ready_status(node, timeout):
-    for _ in range(timeout):
-        if kubecli.get_node_status(node) == "Ready":
+    w = watch.Watch()
+    v1 = client.CoreV1Api()
+    resource_version = get_node_resource_version(node)
+
+    for event in w.stream(
+        v1.list_node,
+        field_selector=f"metadata.name={node}",
+        timeout_seconds=timeout,
+        resource_version=f"{resource_version}",
+    ):
+        conditions = [
+            status
+            for status in event["object"].status.conditions
+            if status.type == "Ready"
+        ]
+        if conditions[0].status == "True":
+            w.stop()
             break
-        time.sleep(3)
-        if kubecli.get_node_status(node) != "Ready":
-            raise Exception("Node condition status isn't Ready")
+        else:
+            logging.info("node status " + str(conditions[0].status))
 
 
-# Wait till node status becomes NotReady
+# Wait until the node status becomes Unknown
 def wait_for_unknown_status(node, timeout):
-    for _ in range(timeout):
-        try:
-            node_status = kubecli.get_node_status(node, timeout)
-            if node_status is None or node_status == "Unknown":
-                break
-        except Exception:
-            logging.error("Encountered error while getting node status, waiting 3 seconds and retrying")
-        time.sleep(3)
-    node_status = kubecli.get_node_status(node, timeout)
-    logging.info("node status " + str(node_status))
-    if node_status is not None and node_status != "Unknown":
-        raise Exception("Node condition status isn't Unknown after %s seconds" % str(timeout))
+    w = watch.Watch()
+    v1 = client.CoreV1Api()
```

Comment:
again, let's move this to https://github.com/chaos-kubox/krkn/blob/main/kraken/kubernetes/client.py so all functionality/setting of CoreV1Api is in that file